src/bin/idl/mod.rs

```diff
@@ -310,7 +297,7 @@ fn instruction(
             )?;
         }
 
-        write!(f, ") {}external", if state { "" } else { "view " })?;
+        write!(f, ") external")?;
```

Comment:
The ts code makes functions view *if* there are return values and there is no mutable account needed. We could do the same.

Reply:
Good point. I'm going to add code to handle this.